blockchain/src/genesis.rs

```diff
@@ -44,13 +45,13 @@ pub fn genesis(keychains: &[KeyChain], amount: i64) -> Vec<Block> {
         let previous = Hash::digest(&"genesis".to_string());
         let base = BaseBlockHeader::new(version, previous, epoch, timestamp);
 
-        let witnesses = keychains
+        let witnesses: BTreeSet<cosi_keys::PublicKey> = keychains
             .iter()
             .map(|p| p.cosi_pkey.clone())
-            .collect::<Vec<cosi_keys::PublicKey>>();
-        let leader = witnesses[0].clone();
+            .collect::<BTreeSet<cosi_keys::PublicKey>>();
```

Comment:
You can omit this type parameter
```suggestion
            .collect();
```